packages/components/src/mobile/bottom-sheet/index.native.js

```diff
@@ -242,32 +242,40 @@ class BottomSheet extends Component {
 
 	onSetMaxHeight() {
 		const { height, width } = Dimensions.get( 'window' );
-		const { safeAreaBottomInset } = this.state;
+		const { safeAreaBottomInset, isFullScreen } = this.state;
 		const statusBarHeight =
 			Platform.OS === 'android' ? StatusBar.currentHeight : 0;
 
+		let maxHeight = height;
+
 		// `maxHeight` when modal is opened along with a keyboard
 		const maxHeightWithOpenKeyboard =
 			0.95 *
-			( Dimensions.get( 'window' ).height -
+			( height -
 				this.keyboardHeight -
 				statusBarHeight -
 				this.headerHeight );
 
-		// On horizontal mode `maxHeight` has to be set on 90% of width
-		if ( width > height ) {
-			this.setState( {
-				maxHeight: Math.min( 0.9 * height, maxHeightWithOpenKeyboard ),
-			} );
-			//	On vertical mode `maxHeight` has to be set on 50% of width
+		// If full screen, use 100% of the inset height
+		// If landscape, use 90% of inset height
+		// If portait, use 50% of inset height
+		if ( isFullScreen ) {
+			maxHeight =
+				height -
+				safeAreaBottomInset -
+				1.2 * this.keyboardHeight -
```

Comment:
I'm not sure how to resolve the `1.2` magic scaler here. When the keyboard is not open, this goes to 0 and has no impact on the bottom sheet height. But I noticed when the keyboard is open the bottom of the menu is cut off on my Pixel 5.